core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java

```diff
@@ -311,7 +311,13 @@ public void run() {
                     while (!(shouldStop.get() || source.exhausted())) {
                         int bytesReceived = inputStream.read(buffer);
 
-                        handler.channelRead(null, Unpooled.wrappedBuffer(buffer, 0, bytesReceived));
+                        int offset = 0;
+                        for (int i = 0; i < bytesReceived; i++) {
+                            if (buffer[i] == '\n' || i == bytesReceived - 1) {
```

Comment:
Let's just extract those two boolean conditions into two variables, `lineBreak`  and `endOfBuffer`.  Maybe even adding a small comment that line break can happen, if the daemon is answering very fast and it means we got multiple responses.